tracing-subscriber/src/registry/sharded.rs

```diff
@@ -97,9 +115,29 @@ impl Registry {
     fn get(&self, id: &Id) -> Option<Guard<'_, DataInner>> {
         self.spans.get(id_to_idx(id))
     }
+
+    pub(crate) fn ref_guard(&self, id: Id) -> LayerGuard<'_> {
+        CLOSE_COUNT.with(|count| {
+            let c = count.get();
+            count.set(c + 1);
+        });
+        LayerGuard {
+            id,
+            registry: &self,
+        }
+    }
 }
 
 thread_local! {
+    // `CLOSE_COUNT` is used to track how many layers have processed a
+    // close event. Once the `CLOSE_COUNT` is 0, the registry knows that
+    // is is safe to remove a span. It does so via the drop on
+    // `LayerGuard`.
+    //
+    // This behavior is needed to enable a Registry-backed Layer to
+    // access span data after the Layer has recieved the `on_close`
+    // callback.
```

Comment:
this comment is good, but i am not sure if the thread local is the right place for it to be...I might move most of this to the method currently called `ref_guard`, and maybe leave a shorter comment here.

Reply:
Done!